app/services/revocation_registry.py

```diff
@@ -309,42 +311,137 @@ async def revoke_credential(
             )
         )
     except ApiException as e:
-        bound_logger.info(
+        bound_logger.exception(
             "An ApiException was caught while revoking credential. The error message is: '{}'.",
             e.reason,
         )
         raise CloudApiException(
             f"Failed to revoke credential: {e.reason}.", e.status
         ) from e
 
-    if not auto_publish_to_ledger:
-        try:
-            rev_reg_record = await get_active_revocation_registry_for_credential(
-                controller=controller,
-                credential_definition_id=credential_definition_id,
-            )
+    bound_logger.info("Successfully revoked credential.")
 
-            await publish_revocation_entry_to_ledger(
-                controller=controller,
-                revocation_registry_id=rev_reg_record.revoc_reg_id,
-                create_transaction_for_endorser=True,
-            )
-        except CloudApiException as e:
-            if e.status_code == 400:
-                bound_logger.info(
-                    "Bad request: Cannot publish revocation entry to ledger: {}",
-                    e.detail,
-                )
-            else:
-                bound_logger.error(e.detail)
-            raise e
-        except Exception as e:
-            bound_logger.exception("Exception caught when revoking credential.")
-            raise e
 
-        await endorser_revoke()
+async def publish_pending_revocations(
+    controller: AcaPyClient, rrid22crid: PublishRevocations
+):
+    """
+        Publish pending revocations
 
-    bound_logger.info("Successfully revoked credential.")
+    Args:
+        controller (AcaPyClient): aca-py client
+        rrid22crid (PublishRevocations): The revocation registry ID.
+
+    Raises:
+        Exception: When the pending revocations could not be published
+
+    """
+    bound_logger = logger.bind(body=rrid22crid)
+
+    try:
+        await controller.revocation.publish_revocations(
+            body=PublishRevocations(rrid2crid=rrid22crid)
+        )
+    except ApiException as e:
+        bound_logger.exception(
+            "An ApiException was caught while publishing pending revocations. The error message is: '{}'.",
+            e.reason,
+        )
+        raise CloudApiException(
+            f"Failed to publish pending revocations: {e.reason}.", e.status
+        ) from e
+
+    bound_logger.info("Successfully published pending revocations.")
+
+
+async def clear_pending_revocations(
+    controller: AcaPyClient, purge: ClearPendingRevocationsRequest
+) -> PublishRevocations:
+    """
+        Clear pending revocations
+
+    Args:
+        controller (AcaPyClient): aca-py client
+        purge (ClearPendingRevocationsRequest): The revocation registry ID.
```

Comment:
Similar update of docstring here. I also prefer the variable name `purge_request` or `purge_pending_request`